### PR TITLE
doc: release-notes: Add 4.0.0 release notes for networking

### PR DESCRIPTION
**Pull request by rlubos**
Add 4.0.0 release notes for networking.

Ethernet && Wi-Fi TBD separately.

### CODE REVIEW

File: doc/releases/release-notes-4.0.rst [hunk 1]
```diff
@@ -537,59 +537,246 @@ Drivers and Sensors
...
+  * Added support for application callback which allows to assign custom addresses
+    in DHCPv4 server.
+
 * DHCPv6:
 
+  * Fixed a bug where system workqueue could be blocked indefinitely by DHCPv6 client.
+
 * DNS/mDNS/LLMNR:
 
+  * Fixed handling of DNS responses encoded with captial letters.
```
Comment (kartben) [inline doc/releases/release-notes-4.0.rst:586]:
```suggestion
  * Fixed handling of DNS responses encoded with capital letters.
```

File: doc/releases/release-notes-4.0.rst [hunk 2]
```diff
@@ -537,59 +537,246 @@ Drivers and Sensors
...
+
+  * Fixed IPv4 destination address setting when using :c:func:`sendmsg` with
+    :kconfig:option:`CONFIG_NET_IPV4_MAPPING_TO_IPV6` option enabled.
+  * Fixed possible unaligned memory access when in :c:func:`net_context_bind`.
+  * Fixed missing NULL pointer check for V6ONLY option read.
+
 * Network Interface:
 
-* OpenThread
+  * Fixed checksum offloading cheks for VLAN interfaces.
```
Comment (kartben) [inline doc/releases/release-notes-4.0.rst:703]:
```suggestion
  * Fixed checksum offloading checks for VLAN interfaces.
```

File: doc/releases/release-notes-4.0.rst [hunk 3]
```diff
@@ -537,59 +537,246 @@ Drivers and Sensors
...
+  * Fixed :c:func:`net_if_send_data` operation for offloaded interfaces.
+  * Fixed needless IPv6 multicast groups joining if IPv6 is disabled.
+  * Added new :c:func:`net_if_ipv4_get_gw` API function.
+  * Fixed compiler warnings when building with ``-Wtype-limits``
+
+* OpenThread:
+
+  * Added support for :kconfig:option:`CONFIG_IEEE802154_SELECTIVE_TXCHANNEL`
+    option in OpenThread radio platform.
+  * Fixed deprecated/preffered IPv6 address state transitions.
```
Comment (kartben) [inline doc/releases/release-notes-4.0.rst:717]:
```suggestion
  * Fixed deprecated/prefered IPv6 address state transitions.
```

File: doc/releases/release-notes-4.0.rst [hunk 4]
```diff
@@ -537,59 +537,246 @@ Drivers and Sensors
...
+  * Fixed socket leak when closing NSOS socket.
+  * Moved socket service library out of experimental.
+  * Added dedicated ``net_socket_service_handler_t`` callback function type for
+    socket services.
+  * Deprecated ``CONFIG_NET_SOCKETS_POLL_MAX``.
+  * Moved ``zsock_poll()`` and ``zsock_select`` implementations into ``zvfs``
+    library.
+  * Removed ``work_q`` parameter from socket service macros as it was no longer
+    used.
+  * Separated native INET sockets implemnetation from socket syscalls so that
```
Comment (kartben) [inline doc/releases/release-notes-4.0.rst:753]:
```suggestion
  * Separated native INET sockets implementation from socket syscalls so that
```

**Review by mmahadevan108 — DISMISSED** [review]

**Review by kartben — CHANGES_REQUESTED** [review]
some typos or spelling mistakes but looking great!

I would also recommend putting the "Added" and "Implemented" items first in each bullet list, as one often wants to quickly skim through the release notes to spot what's new before potentially looking at bug fixes.  As it stands, many of the sections start with what is possibly the least interesting change, like e.g. an assertion being fixed :)

**Comment by rlubos** [comment]
Thanks @kartben, I've fixed the typos and reordered changelog entries as requested.

**Comment by rlubos** [comment]
> As it stands, many of the sections start with what is possibly the least interesting change, like e.g. an assertion being fixed :)

Need to blame contributors here they did not fix issues earlier, the original order was simply based on the commit order in git log :)

**Review by jukkar — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Review by mmahadevan108 — APPROVED** [review]

**Pull request merged**